api/client/request.go

```diff
@@ -57,15 +59,33 @@ type Status struct {
 	ErrorCode int
 }
 
+var (
+	DefaultContextTimeout = 5 * time.Minute
+)
+
 // NewRequest instance
 func NewRequest(client *http.Client, base *url.URL, verb string, version string, authstring, userAgent string) *Request {
```

Comment:
Would it be better to introduce an API breaking change and add context here? That would ensure usage of context, instead of giving the option to do so.

Reply:
I thought about that, I would prefer to avoid a big impact first. We can adjust as we go with this change. Also, since each package has a `DefaultContextTimeout` var, the caller can affect the default just by changing that value.

Reply:
Makes sense :+1: